### PR TITLE
feat(tree): Log telemetry for undo/redo (revert) operations

### PR DESCRIPTION
**Pull request by Josmithr**
Plumbs the necessary logging infrastructure to allow `TreeCheckout` to log telemetry, including related test infra. Adds logging for `revertRevertible` which, for now, logs the "age" of the commit being reverted relative to head.

[AB#5142](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5142)

### CODE REVIEW

File: packages/utils/telemetry-utils/src/mockLogger.ts [hunk 1]
```diff
@@ -259,3 +264,32 @@ function matchObjects(actual: ITelemetryPropertiesExt, expected: ITelemetryPrope
 	}
 	return true;
 }
+
+/**
```
Comment (Josmithr) [inline packages/utils/telemetry-utils/src/mockLogger.ts:268]: For reviewers: if we are happy with the overall testing pattern introduced in this PR, I'll split these changes out into a separate PR.

File: packages/dds/tree/src/shared-tree/treeCheckout.ts [hunk 2]
```diff
@@ -347,6 +349,20 @@ export interface ITreeCheckoutFork extends ITreeCheckout, IDisposable {
 	rebaseOnto(view: ITreeCheckout): void;
 }
 
+/**
+ * Metrics derived from a revert operations.
```
Comment (yann-achard-MS) [inline packages/dds/tree/src/shared-tree/treeCheckout.ts:353]:
```suggestion
 * Metrics derived from a revert operation.
```

**Review by yann-achard-MS — APPROVED** [review]

File: packages/utils/telemetry-utils/src/mockLogger.ts (hunk 1 above)
Comment (yann-achard-MS) [reply packages/utils/telemetry-utils/src/mockLogger.ts:267]: LGTM but you'll likely want to get review from someone else for this part.

Comment (Josmithr) [reply packages/utils/telemetry-utils/src/mockLogger.ts]: Extracted into #21466

File: packages/utils/telemetry-utils/package.json [hunk 3]
```diff
@@ -153,6 +153,9 @@
 			"RemovedFunctionDeclaration_isValidLegacyError": {
 				"forwardCompat": false,
 				"backCompat": false
+			},
+			"ClassDeclaration_MockLogger": {
+				"forwardCompat": false
```
Comment (Josmithr) [inline packages/utils/telemetry-utils/package.json:158]:
```suggestion
```

Comment (alexvy86) [inline packages/utils/telemetry-utils/package.json:158]: This went to the other PR, right?

**Review by alexvy86 — APPROVED** [review]

File: packages/utils/telemetry-utils/package.json (hunk 3 above)
Comment (Josmithr) [reply packages/utils/telemetry-utils/package.json:158]: Urgh, apparently not. I guess I only ported over the code changes :/

**Pull request merged**